File: codes/yujo/739_dailytemperatures_1.ts
function dailyTemperatures(temperatures: number[]): number[] {
  const n = temperatures.length;
  const result = new Array(n).fill(0); // Initialize result array with zeros
  const stack: number[] = []; // Stack to store indices

  for (let i = 0; i < n; i++) {
    // If current temperature is higher than the temperature at stack's top index
    while (
      stack.length > 0 &&
      temperatures[i] > temperatures[stack[stack.length - 1]]
    ) {
      const prevDay = stack.pop()!; // Previous day's index
      result[prevDay] = i - prevDay; // Calculate days difference
    }
    stack.push(i); // Add current index to stack
  }

  return result;
}

File: codes/naeun/739_dailytemperatures_1.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] answer = new int[temperatures.length];

        Stack<Integer> stack = new Stack<>();
        stack.push(0);

        for(int i = 1; i < temperatures.length; i++){
            while(!stack.isEmpty() && temperatures[stack.peek()] < temperatures[i]){
                int prevIdx = stack.pop();
                answer[prevIdx] = i-prevIdx; // length
            }
            stack.push(i);
        }
        
        return answer;
    }
}